sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/ProcedureCatalog.java

```diff
@@ -34,4 +34,9 @@ public interface ProcedureCatalog extends CatalogPlugin {
    * @return the loaded unbound procedure
    */
   UnboundProcedure loadProcedure(Identifier ident);
+
+  /**
+   * List all procedures in the specified database.
+   */
+  Identifier[] listProcedures(String[] namespace);
```

Comment:
Question: what's our story with using varargs in Java?
I see we use `String[]` in `FunctionCatalog` and `String...` in `ViewCatalog`. Given that it will likely be called by Spark, I don't think it makes a difference. Let's just be consistent.

Reply:
I think for methods that need to be implemented by users, `String[]` is better.

Reply:
I see this was updated. Let me resolve.